app/src/backlight.c

```diff
@@ -142,23 +147,44 @@ int zmk_backlight_set_brt(int brt) {
     return zmk_backlight_save_state();
 }
 
-int zmk_backlight_get_brt() { return state.on ? state.brightness : 0; }
-
-int zmk_backlight_toggle() { return zmk_backlight_set_on(!state.on); }
+int zmk_backlight_toggle() { return state.on ? zmk_backlight_off() : zmk_backlight_on(); }
 
-int zmk_backlight_on() { return zmk_backlight_set_on(true); }
+int zmk_backlight_set_brt(uint8_t brightness) {
+    if (brightness > BRT_MAX) {
+        brightness = BRT_MAX;
+    }
 
-int zmk_backlight_off() { return zmk_backlight_set_on(false); }
+    state.brightness = brightness;
+    state.on = (brightness > 0);
 
-int zmk_backlight_inc() {
-    if (!state.on) {
-        return zmk_backlight_set_brt(MAX(state.brightness, CONFIG_ZMK_BACKLIGHT_BRT_STEP));
+    int rc = zmk_backlight_update();
+    if (rc != 0) {
+        return rc;
     }
-    return zmk_backlight_set_brt(state.brightness + CONFIG_ZMK_BACKLIGHT_BRT_STEP);
+
+    return zmk_backlight_save_state();
 }
 
-int zmk_backlight_dec() {
-    return zmk_backlight_set_brt(state.brightness - CONFIG_ZMK_BACKLIGHT_BRT_STEP);
+uint8_t zmk_backlight_calc_brt(int direction) {
+    uint8_t brightness = state.brightness;
+
+    int b = brightness + (direction * CONFIG_ZMK_BACKLIGHT_BRT_STEP);
+    brightness = CLAMP(b, 0, BRT_MAX);
+
+    return brightness;
+}
+
+int zmk_backlight_change_brt(int direction) {
```

Comment:
See: https://github.com/zmkfirmware/zmk/pull/904#discussion_r688176294

> Maybe use "adjust" instead of "change" to better indicate that this changes relative to the existing value, not changes to a new value.